src/jit/gentree.h

```diff
@@ -4224,15 +4227,35 @@ struct GenTreeSIMD : public GenTreeJitIntrinsic
 struct GenTreeHWIntrinsic : public GenTreeJitIntrinsic
 {
     NamedIntrinsic gtHWIntrinsicId;
-
-    GenTreeHWIntrinsic(var_types type, GenTree* op1, NamedIntrinsic hwIntrinsicID, var_types baseType, unsigned size)
-        : GenTreeJitIntrinsic(GT_HWIntrinsic, type, op1, nullptr, baseType, size), gtHWIntrinsicId(hwIntrinsicID)
-    {
-    }
-
-    GenTreeHWIntrinsic(
-        var_types type, GenTree* op1, GenTree* op2, NamedIntrinsic hwIntrinsicID, var_types baseType, unsigned size)
-        : GenTreeJitIntrinsic(GT_HWIntrinsic, type, op1, op2, baseType, size), gtHWIntrinsicId(hwIntrinsicID)
+    int            gtNumArgs; // number of the non-imm arguments
+    int            gtIval;    // the imm8 argument
```

Comment:
> We should be able to unify both and distinguish them

There are different approaches possible, but I would be for using sign bit as indication if it represents `const` (no bit set) or `variable` (bit set)

> Instruction has a fixed, non-user provided, constant

In this case value can be decoded from intrinsic name - we do not need to carry it or decode in importer, this can be done in tablegen

same as above `gtIval` - Ival has `ssize_t` type in `GenTree` node